packages/datetime/src/dateInput.tsx

```diff
@@ -447,10 +464,37 @@ export class DateInput extends AbstractComponent<IDateInputProps, IDateInputStat
             // the page. tabbing forward should *not* close the popover, because
             // focus will be moving into the popover itself.
             this.setState({ isOpen: false });
+        } else if (e.which === Keys.ESCAPE) {
+            this.setState({ isOpen: false });
+            this.inputRef.blur();
         }
         this.safeInvokeInputProp("onKeyDown", e);
     };
 
+    private registerPopoverBlurHandler = () => {
+        const node = ReactDOM.findDOMNode(this);
```

Comment:
i'm not a huge fan of using `findDOMNode`, tend to prefer refs. but this is alright.

Reply:
- [ ] maybe use refs?

Reply:
I just wrapped the popover content since `ReactDayPicker.Props` doesn't have a prop to expose ref.